api/services/gameService.js

```diff
@@ -120,17 +112,20 @@ module.exports = {
       currentMatch: null,
     };
     let { game } = options;
-    const p0Wins = userService.checkWin({ user: game.players[0] });
-    const p1Wins = userService.checkWin({ user: game.players[1] });
+    const [p0, p1] = game.players;
+    const p0Wins = userService.checkWin({ user: p0 });
+    const p1Wins = userService.checkWin({ user: p1 });
     if (p0Wins || p1Wins) {
       res.gameOver = true;
       const gameUpdates = {};
       if (p0Wins) {
         res.winner = 0;
-        gameUpdates.result = GameResult.P0_WINS;
+        gameUpdates.winner = p0.id;
+        gameUpdates.status = GameStatus.FINISHED;
```

Comment:
We can DRY this up by moving these status updates into the above block for where either player wins  since it's the same regardless of which player

Reply:
Nice catch!